### PR TITLE
Zeek 6.1 doc updates

### PR DESCRIPTION
**Pull request by awelzel**

### CODE REVIEW

File: script-reference/statements.rst [hunk 1]
```diff
@@ -456,6 +493,9 @@ Here are the statements that the Zeek scripting language supports.
   * - :zeek:keyword:`add`, :zeek:keyword:`delete`
     - Add or delete elements
 
+  * - :zeek:keyword:`assert`
+    - Runtime sanity checking
```
Comment (bbannier) [inline script-reference/statements.rst:497]: Some consider "sanity check" abelist, maybe go with something like _assertion_ instead. Here and below.

```suggestion
    - Runtime assertion
```

Comment (awelzel) [reply script-reference/statements.rst]: I was not aware, thanks.

File: script-reference/attributes.rst [hunk 2]
```diff
@@ -152,7 +152,42 @@ default value is ``0``.  Example:
...
+        id: conn_id &log;
+        msg: string &log;
+        hidden: count &default=0;  # This is not logged.
+    };
+
+    event zeek_init() {
+        Log::create_stream(LOG, [$columns=Info, $path="example"]);
+    }
+
+Above log stream will have the columns ``ts``, ``id`` and ``msg``.
```
Comment (timwoj) [inline script-reference/attributes.rst:175]:
```suggestion
The log stream above will have the columns ``ts``, ``id`` and ``msg``.
```

File: script-reference/statements.rst [hunk 3]
```diff
@@ -496,6 +536,54 @@ Example:
...
+~~~~~~
+
+.. versionadded:: 6.1
+
+The ``assert`` statement can be used for runtime assertion checks or as a building
+block for a testing framework. It takes an expression ``expr`` of type
+:zeek:see:`bool` and an optional message of type :zeek:see:`string`.
+If ``expr`` at runtime evaluates to ``F``, by default the string representation
+of the expression and the given message, if any, are logged
+via :zeek:see:`Reporter::error`.
```
Comment (timwoj) [inline script-reference/statements.rst:551]:
```suggestion
If ``expr`` at runtime evaluates to ``F``, the string representation
of the expression and the given message, if any, are logged
via :zeek:see:`Reporter::error` by default.
```

File: script-reference/statements.rst [hunk 4]
```diff
@@ -496,6 +536,54 @@ Example:
...
+
+   Zeek's exit code in this example will be ``0``, indicating success.
+   Script errors other than those in a ``zeek_init()`` handler are not
+   reflected in Zeek's exit code.
+
+
+The logging behavior of failing assert statements can be customized using the
+:zeek:see:`assertion_failure` or zeek:see:`assertion_result` hook.
+Using :zeek:see:`break` in either hook allows to suppress the default log
+generation. The :zeek:see:`assertion_result` hook is targeted for testing
```
Comment (timwoj) [inline script-reference/statements.rst:582]:
```suggestion
Using :zeek:see:`break` in either hook allows for suppression of the default log
generation. The :zeek:see:`assertion_result` hook is targeted for testing
```

**Review by timwoj — APPROVED** [review]

**Pull request merged**